pages/how-to-contribute.md

```diff
@@ -4,25 +4,179 @@
+The `constructor` method will inherit the superclass, define the check default customization, and set the code inspector message:
 
-Thank you in advance for contributing and sharing your ideas within our community! We really appreciate this!
+```abap
+METHOD constructor.
+    super->constructor( ).
+
+    settings-prio = <its_priority>.
+    settings-threshold = <its_threshold>.
+    settings-documentation = |{ c_docs_path-checks }<its_documentation>.md|.
+    settings-pseudo_comment = '"#EC <its_pseudo_comment>' ##NO_TEXT.
+    ...
+
+    set_check_message( '<its_message>' ).
+ENDMETHOD.
+```
+
+The `inspect_tokens` method will detect the issue, validate the customizing, and raise the check.  
+Here, you can use the imported `structures` and `statements` to loop the `tokens`.
+
+```abap
+LOOP AT ref_scan_manager->get_tokens( ) ASSIGNING FIELD-SYMBOL(<token>)
+FROM statement-from TO statement-to.
+    ...
+ENDLOOP.
+```
+
+> :bulb: Before start implementing the `inspect_tokens` method, we recommend you to create the unit tests to follow the Test Driven Development (TDD) approach.
+
+The `detect_check_configuration` method validates the check customizing.
+If an empty structure is received, it means the check should not raised.
+
+```abap
+DATA(check_configuration) = detect_check_configuration( error_count = <error_count>
+                                                        statement = <statement_structure> ).
+
+IF configuration IS INITIAL.
+    RETURN.
+ENDIF.
+```
+
+The `raise_error` method raises the check.
+
+```abap
+raise_error( statement_level = statement-level
+             statement_index = index
+             statement_from = statement-from
+             error_priority = configuration-prio ).
+```
+
+The `execute_check` method can be redefined when the check searches for an issue in a non-default statement type. The default types are defined in the `y_check_base`->`execute_check`.
+
+### How to Create the Unit Test
+
+Create a local test class under the global test class created above. It should inherit and implement the abstract methods from the `y_unit_test_base` class.  
+We will use the `y_check_prefer_is_not` check as an example.
+
+> :bulb: We recommend you to keep the test class as simple as possible. You can create multiples local test classes to validate distinct behaviors valid for the check.
+
+```abap
+CLASS ltc_not_is_initial DEFINITION INHERITING FROM y_unit_test_base FOR TESTING RISK LEVEL HARMLESS DURATION SHORT.
+    PROTECTED SECTION.
+        METHODS get_cut REDEFINITION.
+        METHODS get_code_with_issue REDEFINITION.
+        METHODS get_code_without_issue REDEFINITION.
+        METHODS get_code_with_exemption REDEFINITION.
+ENDCLASS.
+```
+
+The `get_cut` method must return an instance of the class under test.
+
+```abap
+METHOD get_cut.
+    result ?= NEW y_check_prefer_is_not( ).
+ENDMETHOD.
+```
+
+The `get_code_with_issue` method must return a snippet of code which raises the check.  
```

Comment:
The code snipped must be syntactical correct.

Reply:
ok. I added the ... as a comment.